File: catalog/samples/wc.swift
/**
 * Sample code using the experimental Swift 3 runtime
 * A word count utility
 * @param payload A string of words to count.
 */
func main(args: [String:Any]) -> [String:Any] {
    if let str = args["payload"] as? String {
        let words = str.components(separatedBy: " ")
        let count = words.count
        print("The message '\(str)' has \(count) words")
        return ["count": count]
    } else {
        return ["error": "You must specify a payload parameter!"]
    }
}

File: tests/dat/actions/wc.js
// Licensed to the Apache Software Foundation (ASF) under one or more contributor
// license agreements; and to You under the Apache License, Version 2.0.

/**
 *  word count utility
 */
function main(params) {
    var str = params.payload.toString();
    var words = str.split(" ");
    var count = words.length;
    console.log("The message '"+str+"' has", count, 'words');
    return { count: count };
}
